internal/bson/document.go

```diff
@@ -34,35 +34,56 @@ const (
 //
 // TODO Remove it.
 type document interface {
-	Map() map[string]any
 	Keys() []string
+	Values() []any
 }
 
 // Document represents BSON Document type.
-//
-// Duplicate fields are not supported yet.
-// TODO https://github.com/FerretDB/FerretDB/issues/1263
 type Document struct {
-	m    map[string]any
-	keys []string
+	fields []field
+}
+
+// field represents a field in the document.
+type field struct {
+	value any
+	key   string
```

Comment:
It's not necessary, just seems right to put the key field first.
```suggestion
	key   string
	value any
```

Reply:
Linter complains about it :/ 
The `fieldalignment` thing